tests/phpunit/tests/post/getPostStatus.php

```diff
@@ -164,4 +164,52 @@ public function data_get_post_status_after_trashing() {
 			array( 'delete-attachment', 'publish', 'publish' ),
 		);
 	}
+
+	/**
+	 * Ensure the `post_states_string` filter works to modify post state output.
+	 *
+	 * @ticket 51403
+	 * @dataProvider data_filter_post_states_string_should_enable_post_state_html_output_modification
+	 *
+	 * @param string $post_state The post state to test.
+	 */
+	public function test_filter_post_states_string_should_enable_post_state_html_output_modification( $post_state ) {
+		$post = get_post( self::$post_ids[ $post_state ] );
+
+		$random_string   = wp_generate_password( 12 );
+		$original_output = _post_states( $post, false );
+		$text_to_append  = '<span class="post-state">, ' . $random_string . '</span>';
```

Comment:
```suggestion
		$original_output = _post_states( $post, false );
		$text_to_append  = '<span class="post-state">, Sample state</span>';
```

Simplify as it's unit test